File: DynapredictAPI/Models/Machine.cs
using System.ComponentModel.DataAnnotations;

namespace DynapredictAPI.Models
{
    public class Machine
    {
        public int Id { get; set; }
        
        [Required]
        public string Name { get; set; } = string.Empty;
        
        [Required]
        public string SerialNumber { get; set; } = string.Empty;
        
        public string Description { get; set; } = string.Empty;
        
        [Required]
        public string Type { get; set; } = string.Empty;
        
        public string Status { get; set; } = "active";
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: dynapredict-frontend/src/types/machine.ts
export interface Machine {
  id: number;
  name: string;
  serialNumber: string;
  description: string;
  type: string;
  status: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface CreateMachineData {
  name: string;
  serialNumber: string;
  description: string;
  type: string;
}

export interface UpdateMachineData {
  name?: string;
  serialNumber?: string;
  description?: string;
  type?: string;
  status?: string;
}
